lib/query_builder.ex

```diff
@@ -12,7 +12,7 @@ defmodule EctoShorts.QueryBuilder do
     builder.create_schema_filter(filter_tuple, query)
   end
 
-  @spec query_schema(Ecto.Query.t) :: Ecto.Schema.t
+  @spec query_schema(Ecto.Queryable.t) :: Ecto.Queryable.t()
```

Comment:
It can have a schema passed in as well and the third function head can return a query.